src/app/companies/companies.component.ts

```diff
@@ -0,0 +1,95 @@
+import {Component, OnInit, ViewChild} from '@angular/core';
+import {CompaniesService} from "./companies.service";
+import {MatSort} from '@angular/material/sort';
+import {MatTableDataSource} from '@angular/material/table';
+import {AuthService} from "../auth/auth.service";
+import {DialogComponent} from "../dialog/dialog.component";
+import {MatDialog} from "@angular/material/dialog";
+import {Company} from "../interfaces/company";
+
+@Component({
+  selector: 'app-companies',
+  templateUrl: './companies.component.html',
+  styleUrls: ['./companies.component.css']
+})
+
+export class CompaniesComponent implements OnInit {
+
+  companies: Company[] = [];
+
+  columns = ['company_name', 'company_address', 'company_email', 'company_contact', 'full_name', 'designation', 'email', 'contact', 'stage', 'signed-up',
+    'contacted', 'rejected', 'accepted'];
+  selectedColumns = this.columns;
+
+  // @ts-ignore
+  @ViewChild(MatSort) sort: MatSort;
+  dataSource = new MatTableDataSource(this.companies);
+  filter = {
+    quick_filter: "",
+    signed_up: {
+      start: null,
+      end: null
+    }
+  };
+
+  renderedData: any;
+
+  constructor(public companiesService: CompaniesService, public authService: AuthService, private dialog: MatDialog) {
+  }
+
+  async ngOnInit() {
+    try {
+      this.companies = await this.companiesService.getCompanies();
+      console.log("Companies: ", this.companies);
+      this.dataSource = new MatTableDataSource<Company>(this.companies);
+      this.dataSource.sort = this.sort;
+      this.dataSource.connect().subscribe(d => this.renderedData = d);
+      this.getDisplayedColumns();
+      //console.log("Role: ", this.authService.getRole());
+    } catch (e) {
+        this.dialog.open(DialogComponent, {
+          data: {
+            type: "error",
+            title: "ERROR",
+            message: e + "Entity:" + this.authService.getEntity() + " Role:"+ this.authService.getRole() + " Email:" + this.authService.getEmail()
```

Comment:
if we can add, lets add line breaks between the different parts of message like 
Entity: BLAH BLAH BLAH
Role: BLAH BLAH BLAH
Might have to deal with a little bit of css to make it work.